homeassistant/components/tuya/models.py

```diff
@@ -270,18 +270,28 @@ class DPCodeEnumWrapper(DPCodeTypeInformationWrapper[EnumTypeData]):
     """Simple wrapper for EnumTypeData values."""
 
     DPTYPE = DPType.ENUM
+    _strict = False
 
     def read_device_status(self, device: CustomerDevice) -> str | None:
         """Read the device value for the dpcode.
 
         Values outside of the list defined by the Enum type information will
         return None.
         """
-        if (
-            raw_value := self._read_device_status_raw(device)
-        ) in self.type_information.range:
-            return raw_value
-        return None
+        if (raw_value := self._read_device_status_raw(device)) is None:
+            return None
+        if raw_value not in self.type_information.range:
```

Comment:
This should **never** happen. This needs to be reported with Tuya honestly...

This should also **not** be solved in quirks, this is a straight up defect.